doc/aspice_swe3_4/example/iceoryx_component/source/example_component/example_base_class.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_BASE_CLASS_HPP
+#define IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_BASE_CLASS_HPP
+
+namespace example
+{
+/// @brief Short description
+/// @code
+///   minimalistic working example which uses all public methods
+/// @endcode
+/// @swcomponent example_component
+void someOtherFreeFunction() noexcept;
+
+/// @brief Base class which is not part of the public API
+/// @details Detailed description
+/// @startuml
+/// 	Alice -> Bob: Authentication Request
+/// 	Bob --> Alice: Authentication Response
+///
+/// 	Alice -> Bob: Another authentication Request
+/// 	Alice <-- Bob: another authentication Response
+/// @enduml
+/// @swcomponent example_component
+template <typename T>
+class ExampleBaseClass
+{
+  public:
+    /// @brief Short description
+    /// @details Detailed description
+    /// @param[in] a Description of input parameter a
+    /// @param[in] b Description of input parameter b
+    ExampleBaseClass(const uint32_t a, const uint32_t b);
+
+    /// @brief Short description
+    void simplerMethod() const noexcept;
+
+    /// @brief Short description
+    uint32_t m_memberVariable{0};
```

Comment:
Could be private or protected (encapsulation). Of course this depends on the use of the class and this is just an example. Personally I advocate having non-private members only in very simple classes without methods or few methods, essentially data bundles (e.g. a point in 3d space). In this case I prefer to omit the m_ prefix for readability. Just a style proposal for us to consider.

Reply:
i will move into private in the next PR